### PR TITLE
#554 - "Remove branch" button is not displayed as expected

### PR DESCRIPTION
**Pull request by LetItRock**
**Changed show "Delete branch" button logic:**
- pull request state should be **closed** or **merged**
- user must have **push permissions** to head repository
- head **reference should exist**

I have tested this change and its working for me... but it would be better if you @alorma will do some testing also...

### CODE REVIEW

**Comment by alorma** [comment]
@juherr ping!

File: app/src/main/java/com/alorma/github/ui/fragment/pullrequest/PullRequestConversationFragment.java [hunk 1]
```diff
@@ -311,10 +314,20 @@ public void onResponseOk(final PullRequestStory pullRequestStory) {
...
+    RepoInfo headRepoInfo = pullRequestStory.item.head.repo.toInfo();
+    GetRepoClient getRepoClient = new GetRepoClient(headRepoInfo);
     GetReferenceClient referenceClient =
-            new GetReferenceClient(issueInfo.repoInfo, pullRequestStory.item.head.ref);
-    referenceClient
+            new GetReferenceClient(headRepoInfo, pullRequestStory.item.head.ref);
+    getRepoClient
             .observable()
+            .flatMap((repo) -> {
+              hasPushPermissionsToHead = repo.permissions.push;
```
Comment (alorma) [inline app/src/main/java/com/alorma/github/ui/fragment/pullrequest/PullRequestConversationFragment.java]: should check if permissions is not null, i get null sometimes

**Comment by juherr** [comment]
If you provide an apk, I will test it on my side.

File: app/src/main/java/com/alorma/github/ui/fragment/pullrequest/PullRequestConversationFragment.java (hunk 1 above)
Comment (LetItRock) [reply app/src/main/java/com/alorma/github/ui/fragment/pullrequest/PullRequestConversationFragment.java]: Ok didn't know about that, I will fix this...

**Comment by alorma** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/gitskarios/Gitskarios/pull-request/559/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=gitskarios/Gitskarios)

**Comment by alorma** [comment]
Great! If you could generate an APK, and send it to @juherr , he can test :P

**Comment by alorma** [comment]
Merge the latest change from develop plz

**Comment by LetItRock** [comment]
Ok. @juherr I have sent APK to you :) Please check your email box julien@herr.fr

**Comment by juherr** [comment]
:+1:  As replied by email, let me few days to test it because I don't have any merged pull-request.

**Comment by LetItRock** [comment]
You can create pull request for my repo changing README.md
https://github.com/LetItRock/test
And I will merge it...

**Comment by juherr** [comment]
Nice try but: "Failed to delete branch" on both refused and merged branches.
Any way to activate logs?

**Comment by LetItRock** [comment]
That means that we get different response from 204... 
https://developer.github.com/v3/git/refs/

**Comment by alorma** [comment]
If you tell me where to test, i will test it later

**Comment by LetItRock** [comment]
@juherr could you check it again with latest APK I have sent?

**Comment by juherr** [comment]
It works now ! Congratulations

**Comment by alorma** [comment]
Merged!

**Comment by LetItRock** [comment]
Great! :) Next time I should be more precise with testing...

**Comment by alorma** [comment]
Doing those PRs was awesome!

**Pull request merged**